contracts/governance/extensions/GovernorTimelockControl.sol

```diff
@@ -117,12 +121,15 @@ abstract contract GovernorTimelockControl is IGovernorTimelock, Governor {
      * @dev Overridden execute function that run the already queued proposal through the timelock.
      */
     function _execute(
-        uint256 /* proposalId */,
+        uint256 proposalId,
         address[] memory targets,
         uint256[] memory values,
         bytes[] memory calldatas,
         bytes32 descriptionHash
     ) internal virtual override {
+        // cleanup for refund
+        delete _timelockIds[proposalId];
```

Comment:
I would feel more comfortable deleting the id after the execute call for cases of reentrancy.

Reply:
I was thinking the opposite: doing all the local changes before having an external call and a possible reentrancy

Reply:
I see the reasoning... But in this case the local change is not "mark it as done", it is "erase the way to know that it's already scheduled".

Reply:
Its more than that ... by cleaning that we are changing the way `state(uint256)` behaves. 

I'm not saying I'm doing it write, but we need to consider what we want the state to be is a reentrancy call happens. At first glance, it feels more correct to return Queued than Succeeded... so we should probably clean after.

Reply:
I still would feel more comfortable deleting after the call.